### ISSUE TITLE
Support dynamic reloading of credentials config file

### ISSUE THREAD

**Issue by jtackaberry** [rqlite/rqlite#1553]
User management, especially password rotation, is a common operational activity.  Currently the credentials config file is only read once at startup, which means rotating a password requires a rolling restart of the cluster.

This FR is a minor quality-of-life improvement to support dynamic reloading of the user credentials config file and handle password changes, user additions, removals, and permissions changes.

This could be done using [fsnotify](https://github.com/fsnotify/fsnotify), which would be fairly robust but requires sourcing a new dependency, or by simply using `os.Stat()` and polling for mtime changes every few seconds.  Regardless of the method, the implementation should tolerate the path provided to rqlited being a symlink, and follow the symlink in detecting changes.

**Comment by pn03** [comment]
Hey @jtackaberry is this still required ?

If so I would like to work on this.

**Comment by otoolep** [comment]
Yes, it is still required. I implemented it for TLS certs, but the standard library helped as it had a callback.

fsnotify has some OS-specific gotchas (I think) so building something straightforward using the standand library is better. A simple type which polls a file once a second (should be configurable) and then reloads the info would be best.

However I would suggest you first refactor the credentials code so it has a TLS-like callback, just like the standard library (see issue #1554). First add that, then add the file watcher, which supports such a callback.

**Comment by otoolep** [comment]
See https://github.com/rqlite/rqlite/blob/master/internal/rtls/reloader.go to see how it works for TLS. This works because the standard library supports a callback interface. A similar interface could be added to the Credential Store.

**Comment by pn03** [comment]
@otoolep Thanks for all the information !
